src/seedu/addressbook/Main.java

```diff
@@ -105,15 +106,23 @@ private void recordResult(CommandResult result) {
      * @param command user command
      * @return result of the command
      */
-    private CommandResult executeCommand(Command command)  {
-        try {
+    private CommandResult executeCommand(Command command)
+    {
+        try
+        {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
-        } catch (Exception e) {
+        }
+        catch (StorageOperationException e) {
             ui.showToUser(e.getMessage());
```

Comment:
As discussed in tutorial, you should provide useful message to the user instead of stack trace.